src/nvim/ex_cmds.c

```diff
@@ -6117,5 +6123,6 @@ void do_inc_sub(exarg_T *eap)
     close_windows(livebuf, false);
     close_buffer(NULL, livebuf, DOBUF_WIPE, false);
   }
+  curbuf->b_p_ul = b_p_ul_save;
```

Comment:
minor nitpick: I would have placed this right after `restore_search_patterns` which also is a "restore" line. As now it is placed in the middle of the redraw logic.